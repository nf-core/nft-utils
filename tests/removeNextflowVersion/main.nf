workflow {

  Channel.of(
      """
    Workflow:
        Pipeline: 1.0.0
        Nextflow: ${workflow.nextflow.version}
    """.stripIndent().trim()
    )
    .collectFile(storeDir: "${params.outdir}/pipeline_info", name: 'nf_core_pipeline_software_mqc_versions.yml', sort: true, newLine: true)
}
