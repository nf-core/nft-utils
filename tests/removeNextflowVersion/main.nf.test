nextflow_pipeline {

    name "Test removeNextflowVersion"
    script "./main.nf"
    tag "removeNextflowVersion"

    test("removeNextflowVersion") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          assert snapshot(removeNextflowVersion("$outputDir/pipeline_info/nf_core_pipeline_software_mqc_versions.yml")).match()
        }
    }

    test("removeNextflowVersion with wildcard") {
        when {
            params {
                outdir = "$outputDir"
            }
        }

        then {
          assert snapshot(removeNextflowVersion("$outputDir/pipeline_info/*_versions.yml")).match()
        }
    }
}
