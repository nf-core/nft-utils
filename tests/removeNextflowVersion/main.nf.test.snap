{
    "removeNextflowVersion": {
        "content": [
            {
                "Workflow": {
                    "Pipeline": "1.0.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.04.4"
        },
        "timestamp": "2024-09-27T19:17:42.066036"
    },
    "removeNextflowVersion with wildcard": {
        "content": [
            {
                "Workflow": {
                    "Pipeline": "1.0.0"
                }
            }
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-06-05T14:20:18.078399064"
    }
}